src/main/java/roomescape/domain/Reservation.java

```diff
@@ -18,12 +18,12 @@ public Reservation(Long id, String name, LocalDate date, LocalTime time) {
     }
 
     public Reservation(String name, LocalDate date, LocalTime time) {
+        validateDate(date);
         this.id = null;
         this.name = name;
         this.date = date;
         this.time = time;
     }
```

Comment:
생성 단계에서 검증이 이루어지네요.

현재 Reservation에는 2개의 생성자가 있는데...
날짜에 대한 정책은 해당 생성자를 통해서 만드는 경우에만 지켜질 것 같아요.

`validateDate()`를 주 생성자로 옮기고 `this()`를 통해 재사용 할 수도 있어요.
```suggestion
// 주 생성자
public Reservation(Long id, String name, LocalDate date, LocalTime time) {
    validateDate(date);
    this.id = id;
    this.name = name;
    this.date = date;
    this.time = time;
}

// 부 생성자
public Reservation(String name, LocalDate date, LocalTime time) {
    this(null, name, date, time);
}
```